cmd/makedocs/main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+
+	"github.com/PieselBois/kfulint/lint"
+)
+
+const (
+	docsPath      = "../../docs/"
+	checkersPath  = docsPath + "checkers/"
+	templatesPath = docsPath + "templates/"
+)
+
+type checker struct {
+	Name         string
+	Description  string
+	Experimental bool
+}
+
+func main() {
+	tmpl, err := template.ParseFiles(templatesPath + "overview.md.tmpl")
+	if err != nil {
+		log.Fatal(err)
+	}
+	checkers := []checker{}
+	for _, r := range lint.RuleList() {
+		desc, err := getDesc(r.Name())
+		if err != nil {
+			log.Fatal(r.Name())
+		}
+		checkers = append(checkers, checker{
+			Name:         r.Name(),
+			Experimental: r.Experimental(),
+			Description:  desc,
+		})
+	}
+	file, err := os.OpenFile(docsPath+"overview.md", os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = tmpl.Execute(file, struct {
+		Checkers []checker
+	}{
+		Checkers: checkers,
+	})
```

Comment:
You can use `bytes.Buffer` for `tmpl.Execute` and then do `ioutil.WriteFile`.
This way, you don't need to mess around `OpenFile` and it's flags, you don't need to close file either.